enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,501 @@
+ExternalDNS requires different permissions for each cloud provider, as documented in the
+[tutorial section of the ExternalDNS docs](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials).
+In general, ExternalDNS needs proper permissions to view, update, and delete DNS resource records in the relevant DNS zones.
+
+Support for OpenShift clusters running AWS that
+[leverage STS](https://docs.openshift.com/container-platform/4.7/authentication/managing_cloud_provider_credentials/cco-mode-sts.html#sts-mode-installing-manual-run-installer)
+for AWS credentials should also be added to the ExternalDNS Operator. While ExternalDNS has some level of support for AWS STS, there is a
+[known caveat upstream](https://github.com/kubernetes-sigs/external-dns/issues/1944) that may affect OCP.
+
+In the event that invalid provider credentials are passed to ExternalDNS, a `ValidCredentials` status condition could be set to false on the
+ExternalDNS Operator CR. On OpenShift, this status condition could be reflected by the ExternalDNS ClusterOperator resource
+(should the ExternalDNS Operator be included in the Core OCP product).
+
+Outside of OpenShift, a user could set the `providerCredentials` field to be an object reference to a resource that holds their cloud credentials,
+such as a secret or configmap in an arbitrary namespace.
+
+By default, the operator would assume the cluster's platform is the desired DNS Provider. The operator would allow a user to specify
+an alternative External DNS Provider, such as BlueCat DNS. The ExternalDNS Operator would be able to create multiple ExternalDNS deployments
+to run in concert, one for each ExternalDNS Operator Custom Resource (similar to how the OpenShift [Ingress Operator](https://github.com/openshift/cluster-ingress-operator)
+consumes one or more Ingress Controller resources). The `domains` field would be used to filter source resources according to the requested DNS hostname.
+If an ExternalDNS Operator Custom Resource specifies multiple DNS Zones, multiple instances of ExternalDNS would be deployed within the same deployment accordingly.
+
+By default, ExternalDNS would create DNS records for all instances of the given Source resources. A union type would be defined to allow for
+source-specific parameters. In this example, ExternalDNS would create DNS Records for Route resources that match the label (read: annotation, more on this later)
+`foo: bar`, and would ignore the ExternalDNS Host Target Override Annotation (See [Implementation Details](#implementation-details) for more details about ExternalDNS
+annotations). This example would also instruct ExternalDNS to create DNS Records for CR instances of type `dnsrecords` (not a part of OCP 4.9 implementation).
+Finally, this example would instruct ExternalDNS to create DNS Records in the given Private zone for Service resource instances in namespace `foo`.
```

Comment:
"Records", "See", and "Private" can be lowercase.  My memory is bad, and by the time I reach this paragraph, the example has scrolled off the screen, and I've figuratively and literally lost sight of it; two things could help: (1) an introductory sentence at the start of this section, such as, "The following example CR will be used throughout this section to explain the proposed CRD", and (2) a gentle reminder of where the example is when referencing it.  
```suggestion
source-specific parameters. For the example ExternalDNS CR above, the ExternalDNS Operator would create DNS records for Route resources that match the label (read: annotation, more on this later)
`foo: bar`, and would ignore the ExternalDNS Host Target Override Annotation (see [Implementation Details](#implementation-details) for more details about ExternalDNS
annotations). This example would also instruct ExternalDNS to create DNS records for CR instances of type `dnsrecords` (not a part of OCP 4.9 implementation).
Finally, this example would instruct ExternalDNS to create DNS records in the given private zone for Service resource instances in namespace `foo`.
```
By the way, do you prefer "the ExternalDNS operator" (because that's what it is), "the ExternalDNS Operator", or "ExternalDNS Operator" (without an article because we're using it as a name)?

Reply:
Great suggestions! I've gone ahead and settled on `the ExternalDNS operator`. I've also fixed all of the typos you highlighted, as well as added the requested details from (1) and (2). Thanks!